test/__init__.py

```diff
@@ -97,21 +98,30 @@ def _is_unreachable_err(err):
 
     @functools.wraps(test)
     def wrapper(*args, **kwargs):
-        msg = "Can't run {name} because the network is unreachable".format(
-            name=test.__name__)
-        try:
+        global _requires_network_has_net
+
+        if _requires_network_has_net is None:
+            try:
+                sock = socket.create_connection((TARPIT_HOST, 80), 0.0001)
+                sock.close()
+                _requires_network_has_net = True
+            except socket.timeout:
+                _requires_network_has_net = True
+            except socket.error as e:
+                # This test needs an initial network connection to attempt the
+                # connection to the TARPIT_HOST. This fails if you are in a place
+                # without an Internet connection, so we skip the test in that case.
```

Comment:
So this is a good comment that is now needlessly buried in a big method. Can you hoist this text into the docstring for `requires_network`?